src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -227,31 +215,48 @@ internal abstract class AbstractRefactoringHelpersService : IRefactoringHelpersS
                 while (true);
             }
 
-            if (tokenToLeft != default)
+            // there could be multiple (n) tokens to the left if first n-1 are Empty -> iterate over all of them
+            while (tokenToLeft != default)
             {
                 var leftNode = tokenToLeft.Parent;
                 do
                 {
                     // Consider either a Node that is:
                     // - Ancestor Node of such Token as long as their span ends on location (it's still on the edge)
-                    var acceptedNode = extractNodes(leftNode, syntaxFacts).FirstOrDefault(predicate);
-                    if (acceptedNode != null)
-                    {
-                        return acceptedNode;
-                    }
+                    AddNonHiddenCorrectTypeNodes(ExtractNodesSimple(leftNode, syntaxFacts), relevantNodesBuilder, cancellationToken);
 
                     leftNode = leftNode.Parent;
-                    if (leftNode == null || leftNode.GetLastToken().Span.End != location)
+                    if (leftNode == null || !(leftNode.GetLastToken().Span.End == location || leftNode.Span.End == location))
                     {
                         break;
                     }
                 }
                 while (true);
+
+                // as long as current tokenToLeft is empty -> its previous token is also tokenToLeft
+                tokenToLeft = tokenToLeft.Span.IsEmpty
+                    ? tokenToLeft.GetPreviousToken(includeZeroWidth: true)
+                    : default;
+
+            }
+
+            // If the wanted node is an expression syntax -> traverse upwards even if location is deep within a SyntaxNode
+            // ArgumentSyntax isn't an "expression" syntax but we want to treat it as such
+            if (typeof(TSyntaxNode).IsSubclassOf(typeof(TExpressionSyntax)) || typeof(TSyntaxNode) == typeof(TArgumentSyntax))
```

Comment:
what's teh 'argument' scenario?  which feature needs that.

Reply:
The add addNameToArguement and addNameToTuple. In both cases the wanted node is ArgumentSyntax.

Reply:
can't those features ask for teh relevant expressions.  those will then naturally walk upwards (since you get a lists back).  Then they just ask fort he first of those that is parented by an argument?

Note: i don't feel strongly about this... but it just seems nice to *very very very* much limit the walking to expressions, and allow the *feature* to decide what to do with the results.

Reply:
The problem is that the caller has no way of knowing/influencing the returned node. E.g. `TryGetNodes<LocalDeclaration>` for `var a = 2+[||]3;`. In this case the call would return just the declaration because it climbed. But we didn't want to climb. 

And we also didn't know we climbed / couldn't filter on it because there's no `Func<Node, bool> climbUntil` and the only returned information are the resulting nodes, in this case the one. 

Essentially these instances become indistinguishable ` for `var a = 2+[||]3;` and ` for `var a[||] = 2+3;`

Reply:
Hmm, that should work!